scripts/zones/Northern_San_dOria/npcs/Ailbeche.lua

```diff
@@ -11,26 +11,39 @@ require("scripts/globals/settings");
 require("scripts/globals/keyitems");
 require("scripts/globals/titles");
 require("scripts/globals/quests");
+
 require("scripts/globals/shop");
 -----------------------------------
 
-function onTrade(player,npc,trade)
-    -- "Flyers for Regine" conditional script
-    local FlyerForRegine = player:getQuestStatus(SANDORIA,FLYERS_FOR_REGINE);
+-- Helper constants in order to keep it straight about what part
+-- of the quest we are working on
+BOYS_DREAM_WILL_BEGIN_ACCEPTING_BUG = 3
+BOYS_DREAM_HAS_TRADED_BUG = 4
+BOYS_DREAM_VISIT_ZALDON_FOR_BOOTS_TRADED_FISH = 5
 
+function onTrade(player,npc,trade)
     if (player:getQuestStatus(SANDORIA, FATHER_AND_SON) == QUEST_COMPLETED and player:getVar("returnedAilbecheRod") ~= 1) then
         if (trade:hasItemQty(17391,1) == true and trade:getItemCount() == 1) then
             player:startEvent(61); -- Finish Quest "Father and Son" (part2) (trading fishing rod)
+            return
```

Comment:
there shouldn't be a return here it will kill the event you just started and put an error in the console log for the map server

Reply:
Why is this the case then? Judging by how it looks, one would think it's an async operation.

Is there a place in the C++ code I can read to understand the flow better?